edx_event_bus_kafka/config.py

```diff
@@ -0,0 +1,59 @@
+"""
+Configuration loading and validation.
+"""
+
+from typing import Optional
+
+import warnings
+import attr
+from confluent_kafka.schema_registry import SchemaRegistryClient
+from django.conf import settings
+
+
+@attr.s
+class CommonConfig:
+    settings = attr.ib(type=dict)
+    
+
+def create_schema_registry_client() -> Optional[SchemaRegistryClient]:
+    """
+    Create a schema registry client from common settings.
+    """
+    url = getattr(settings, 'EVENT_BUS_KAFKA_SCHEMA_REGISTRY_URL', None)
+    if url is None:
+        warnings.warn("Cannot configure event-bus-kafka: Missing setting EVENT_BUS_KAFKA_SCHEMA_REGISTRY_URL")
+        return None
+
+    key    = getattr(settings, 'EVENT_BUS_KAFKA_SCHEMA_REGISTRY_API_KEY', '')
+    secret = getattr(settings, 'EVENT_BUS_KAFKA_SCHEMA_REGISTRY_API_SECRET', '')
+
+    return SchemaRegistryClient({
+        'url': url,
+        'basic.auth.user.info': f"{key}:{secret}",
+    })
+
+def load_common_settings(settings) -> Optional[dict]:
```

Comment:
I don't know that the indirection is needed here -- can we just use the imported `settings` directly?

Reply:
when I tried to do that it said something along the lines of settings is used before declaration, possibly having to do with multiple files which makes settings not actually import when being called in the load_common_settings function

Reply:
See other (newer) comment -- I had never actually run the code on that branch, so I hadn't noticed that I was both declaring  a local `settings` in the function but also using `from django.conf import settings` (which makes a global binding of `settings`). The solution is to declare a `base_settings` (or something) at which point you can remove the `settings` parameter.